src/services/services.ts

```diff
@@ -269,20 +275,36 @@ namespace ts {
                 scanner.setText((sourceFile || this.getSourceFile()).text);
                 children = [];
                 let pos = this.pos;
+                const useJSDocScanner =

```

Comment:
`JSDocScanner` limits scanning to several allowed characters, while disallowed characters are all scanned as unknown tokens. JSDoc nodes include JSDoc type nodes and JSDoc type expression nodes, and if the current position is inside a JSDoc type expression, it should allow all characters, because that part is essentially the same as code. For example, the `<` and `>` are not allowed in JSDoc tokens, they are fine when used with generics in JSDoc type expression.